internal/app/repository/db.go

```diff
@@ -0,0 +1,120 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/vovanwin/shorter/internal/app/config"
+	"github.com/vovanwin/shorter/internal/app/model"
+	"log"
+	"os"
+)
+
+type DB struct {
+	pool   *pgxpool.Pool
+	Config config.Config
+}
+
+func NewDB(pool *pgxpool.Pool) *DB {
+	return &DB{
+		pool: pool,
+	}
+}
+
+func (m *DB) GetLink(code string) (model.URLLink, error) {
```

Comment:
```suggestion
func (m *DB) GetLink(ctx context.Context, code string) (*model.URLLink, error) {
```
Не надо шунтировать контекст. Если взята функция, которая с ним работает, надо пробрасывать ей контекст от запроса, тогда в его использовании будет смысл.

А возвращать лучше ссылку на модель, чтобы не создавать копии в памяти.

Эти замечания касаются всех функций из пакета.